ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Mock/Block.hs

```diff
@@ -0,0 +1,348 @@
+{-# LANGUAGE DeriveFunctor         #-}
+{-# LANGUAGE DeriveGeneric         #-}
+{-# LANGUAGE FlexibleContexts      #-}
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE RankNTypes            #-}
+{-# LANGUAGE RecordWildCards       #-}
+{-# LANGUAGE StandaloneDeriving    #-}
+{-# LANGUAGE TypeFamilies          #-}
+
+-- | Simple block to go with the mock ledger
+--
+-- None of the definitions in this module depend on, or even refer to, any
+-- specific consensus protocols.
+module Ouroboros.Consensus.Ledger.Mock.Block (
+    SimpleBlock
+  , SimpleBlock'(..)
+  , SimpleHeader
+  , SimpleHeader'(..)
+  , SimpleStdHeader(..)
+  , SimpleBody(..)
+    -- * Working with 'SimpleBlock'
+  , mkSimpleHeader
+  , matchesSimpleHeader
+    -- * 'UpdateLedger'
+  , LedgerState(..)
+  , LedgerConfig(..)
+  , updateSimpleLedgerState
+  , genesisSimpleLedgerState
+    -- * 'ApplyTx' (mempool support)
+  , GenTx(..)
+    -- * Crypto
+  , SimpleCrypto
+  , SimpleStandardCrypto
+  , SimpleMockCrypto
+    -- * Serialisation
+  , encodeSimpleHeader
+  , decodeSimpleHeader
+  ) where
+
+import qualified Codec.CBOR.Decoding as CBOR
+import qualified Codec.CBOR.Encoding as CBOR
+import           Codec.Serialise (Serialise (..))
+import           Control.Monad.Except
+import           Data.FingerTree (Measured (..))
+import           Data.Typeable (Typeable)
+import           GHC.Generics (Generic)
+
+import           Ouroboros.Network.Block
+
+import           Ouroboros.Consensus.Crypto.Hash
+import           Ouroboros.Consensus.Ledger.Abstract
+import           Ouroboros.Consensus.Ledger.Mock.Address
+import           Ouroboros.Consensus.Ledger.Mock.State
+import           Ouroboros.Consensus.Ledger.Mock.UTxO
+import           Ouroboros.Consensus.Mempool.API
+import           Ouroboros.Consensus.Util ((.:))
+import           Ouroboros.Consensus.Util.Condense
+
+{-------------------------------------------------------------------------------
+  Definition of a block
+
+  The primed versions allow to vary the @ext@ parameter independently of the
+  previous block hash.
+-------------------------------------------------------------------------------}
+
+type SimpleBlock  c ext = SimpleBlock'  c ext ext
+type SimpleHeader c ext = SimpleHeader' c ext ext
+
+data SimpleBlock' c ext ext' = SimpleBlock {
+      simpleHeader :: SimpleHeader' c ext ext'
+    , simpleBody   :: SimpleBody
+    }
+  deriving (Generic, Show, Eq)
+
+data SimpleHeader' c ext ext' = SimpleHeader {
```

Comment:
Add comment: the two type parameters allow us to vary the extension used in the previous block (`simplePrev`) independently from the extension used in this block (`simpleHeaderExt`). This is essential in `forgeExt`.